src/Tasks/Microsoft.NET.Build.Tasks/ResolveReadyToRunCompilers.cs

```diff
@@ -188,13 +206,13 @@ private static bool ExtractTargetPlatformAndArchitecture(string runtimeIdentifie
             platform = null;
             architecture = default;
 
+            // This will split RID like "linux-musl-arm64" into "linux-musl" and "arm64" components
             int separator = runtimeIdentifier.LastIndexOf('-');
             if (separator < 0)
             {
                 return false;
             }
 
-            platform = runtimeIdentifier.Substring(0, separator).ToLowerInvariant();
             string architectureStr = runtimeIdentifier.Substring(separator + 1).ToLowerInvariant();
 
             switch (architectureStr)
```

Comment:
This architecture switch should probably also handle armel.

Reply:
Under the current design it is not possible to have `"armel"` here.